src/init/starship.ps1

```diff
@@ -21,133 +23,130 @@ function Get-Arguments {
+
+            $arguments += "--cmd-duration=$($duration)"
         }
-    } catch {}
-
-	# Get arguments for starship prompt
-	$arguments = Get-Arguments
-
-    # Invoke Starship
-    Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
-
-    # Propagate the original $LASTEXITCODE from before the prompt function was invoked.
-    $global:LASTEXITCODE = $origLastExitCode
-
-    # Propagate the original $? automatic variable value from before the prompt function was invoked.
-    #
-    # $? is a read-only or constant variable so we can't directly override it.
-    # In order to propagate up its original boolean value we will take an action
-    # which will produce the desired value.
-    #
-    # This has to be the very last thing that happens in the prompt function
-    # since every PowerShell command sets the $? variable.
-    if ($global:? -ne $origDollarQuestion) {
-        if ($origDollarQuestion) {
-             # Simple command which will execute successfully and set $? = True without any other side affects.
-            1+1
-        } else {
-            # Write-Error will set $? to False.
-            # ErrorAction Ignore will prevent the error from being added to the $Error collection.
-            Write-Error '' -ErrorAction 'Ignore'
+
+        $arguments += "--status=$($lastExitCodeForPrompt)"
+
+        # Invoke Starship
+        Invoke-Native -Executable ::STARSHIP:: -Arguments $arguments
+
+        # Propagate the original $LASTEXITCODE from before the prompt function was invoked.
+        $global:LASTEXITCODE = $origLastExitCode
+
+        # Propagate the original $? automatic variable value from before the prompt function was invoked.
+        #
+        # $? is a read-only or constant variable so we can't directly override it.
+        # In order to propagate up its original boolean value we will take an action
+        # which will produce the desired value.
+        #
+        # This has to be the very last thing that happens in the prompt function
+        # since every PowerShell command sets the $? variable.
+        if ($global:? -ne $origDollarQuestion) {
+            if ($origDollarQuestion) {
+                 # Simple command which will execute successfully and set $? = True without any other side affects.
+                1+1
+            } else {
+                # Write-Error will set $? to False.
+                # ErrorAction Ignore will prevent the error from being added to the $Error collection.
+                Write-Error '' -ErrorAction 'Ignore'
+            }
         }
-    }
 
-}
+    }
 
-# Get arguments for starship continuation prompt
-$arguments = Get-Arguments
-$arguments += "--continuation"
+    # Invoke Starship and set continuation prompt
+    Set-PSReadLineOption -ContinuationPrompt (
+        Invoke-Native -Executable ::STARSHIP:: -Arguments @(
+            "prompt",
+            "--continuation"
+        )
+    )
```

Comment:
Please move this after `STARSHIP_SHELL` and `STARSHIP_SESSION_KEY`, or the continuation prompt could get escaped for a different shell.